USER_GUIDE.md

```diff
@@ -368,4 +368,51 @@ The plugin defaults to `appengine-web.xml` based deployment if your project cont
 file. If your project also has an `src/main/appengine/app.yaml` file and you wish to use that, you may temporarily move the
 `appengine-web.xml` file to a different location before deploying.
 
+### How do I deploy a thin jar to App Engine?
+
+You can do this if you have an `app.yaml` based project and you have a custom `entrypoint` configured to start it.
+
+You will need to configure the stage goal to include your project's dependencies. You could use something like the
+[`maven-dependency-plugin:copy-dependency`](https://maven.apache.org/plugins/maven-dependency-plugin/copy-dependencies-mojo.html)
+goal to copy dependencies into a directory and then configure that directory to be staged with your application using
+the `extraFilesDirectories` parameter in the appengine plugin. For example:
+
+```xml
+<plugin>
+  <artifactId>maven-dependency-plugin</artifactId>
+  <executions>
+    <execution>
+      <phase>package</phase>
+      <goals>
+        <goal>copy-dependencies</goal>
+      </goals>
+      <configuration>
+        <outputDirectory>${project.build.directory}/dependencies</outputDirectory>
+      </configuration>
+    </execution>
+  </executions>
+</plugin>
+<plugin>
+  <groupId>com.google.cloud.tools</groupId>
+  <artifactId>appengine-maven-plugin</artifactId>
+  <version>2.1.0</version>
+  <configuration>
+    <!-- your config -->
+    <extraFilesDirectories>
+      <extraFilesDirectory>${project.build.directory}/dependencies</extraFilesDirectory>
+    </extraFilesDirectories>
+  </configuration>
+</plugin>
+```
+
+Then when you run `mvn package appengine:deploy` the dependencies are copied and included as part of the deployment.
+
+Since the default entrypoint assumes a fat jar, you must define a custom entrypoint to run this application. In this
+example we might have an `app.yaml` that looks like:
+
+```
+runtime: java11
+entrypoint: java -Xmx64m -cp "*" com.example.MyMainClass
```

Comment:
Is `-Xmx64m` recommended for App Engine? If not, remove it?

Reply:
That's a good point, it was in the example on the appengine docs, so I used it.